docs/vector_store.ipynb

```diff
@@ -1,79 +1,356 @@
+    "index_config = HNSWConfig(\n",
+    "    name=\"my_vector_index\", distance_strategy=DistanceStrategy.COSINE, vector_size=128\n",
+    ")\n",
+    "\n",
+    "# Initialize/create the vector store index\n",
+    "RedisVectorStore.init_index(client=redis_client, index_config=index_config)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Prepare Documents\n",
+    "\n",
+    "Text needs processing and numerical representation before interacting with a vector store. This involves:\n",
+    "\n",
+    "* Loading Text: The TextLoader obtains text data from a file (e.g., \"state_of_the_union.txt\").\n",
+    "* Text Splitting: The CharacterTextSplitter breaks the text into smaller chunks for embedding models."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain.text_splitter import CharacterTextSplitter\n",
+    "from langchain_community.document_loaders import TextLoader\n",
+    "\n",
+    "loader = TextLoader(\"./state_of_the_union.txt\")\n",
+    "documents = loader.load()\n",
+    "text_splitter = CharacterTextSplitter(chunk_size=1000, chunk_overlap=0)\n",
+    "docs = text_splitter.split_documents(documents)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Add Documents to the Vector Store\n",
+    "\n",
+    "After text preparation and embedding generation, the following methods insert them into the Redis vector store."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Method 1: Classmethod for Direct Insertion\n",
+    "\n",
+    "This approach combines embedding creation and insertion into a single step using the from_documents classmethod:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_community.embeddings.fake import FakeEmbeddings\n",
+    "\n",
+    "embeddings = FakeEmbeddings(size=128)\n",
+    "redis_client = redis.from_url(\"redis://127.0.0.1:6379\")\n",
+    "rvs = RedisVectorStore.from_documents(\n",
+    "    docs, embedding=embeddings, client=redis_client, index_name=\"my_vector_index\"\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Method 2: Instance-Based Insertion\n",
+    "This approach offers flexibility when working with a new or existing RedisVectorStore:\n",
+    "\n",
+    "* [Optional] Create a RedisVectorStore Instance: Instantiate a RedisVectorStore object for customization. If you already have an instance, proceed to the next step.\n",
+    "* Add Text with Metadata: Provide raw text and metadata to the instance. Embedding generation and insertion into the vector store are handled automatically."
```

Comment:
"Embedding, generation, and insertion"

Reply:
well - embedding is a noun here :-) think of it as "vector" so this should read as "vector generation and insertion. "